### ISSUE TITLE
[FEATURE REQUEST] Dark mode UI

### ISSUE THREAD

**Issue by Vissova** [Duude92/hammertime#35]
**Is your feature request related to a problem? Please describe.**
A dark mode would be slightly easier on the eyes.

**Describe the solution you'd like**
Add a drop-down box in the settings to change between Light (default) and Dark modes.

**Describe alternatives you've considered**
There are no available alternatives built into the editor.

**Additional context**
As an example, this is the dark mode in TrenchBroom:
![Untitled](https://github.com/user-attachments/assets/b09f74c6-2382-47b4-8585-94cd082b0288)

**Comment by TarekLP** [comment]
Isn't this already a thing in hammertime? Or am I dumb rn

**Comment by Duude92** [comment]
Previously I made a test build with black-mode, but since then I never updated it.
The main problem is that application built over WindowsForms libnrary, which doesn't support theming. I used couple tricks to update the colors of windows, but obviously it looked horible, bc there are a lot of components, which need to be themed manually, So I just continued to work on features/fixes.
I have local git backup, so I can recover branch, if anyone wants to try maintain it. But since I have issues with electricity - I dont have enough time to maintain editor, not just this feature.

**Comment by Duude92** [comment]
No i'm not, just have troubles with electricity, so theming is on the other side of priority list

**Comment by Duude92** [comment]
No, i'm not, but have troubles with electricity, so most feature requests are at the other end of the priority list rn.

**Comment by TarekLP** [comment]
Alright then, as always if ya need help with anything, I'll be mental support or smth

**Comment by Vissova** [comment]
Thank you for the response, and no worries. I'm just glad this project is being worked on at all!

**Comment by TarekLP** [comment]
> ![image](https://github.com/user-attachments/assets/f72286e9-c4a8-480e-b944-39b74f2b8a0d)
> Here is I recovered branch, and, thats why I stopped working on it: it has lots of custom-made ui components, which need to be found and reworked to use theming

Itll take time won't it?

**Comment by Duude92** [comment]
Publish implementation, fixed couple bugs found before, now it looks like that:
![image](https://github.com/user-attachments/assets/2428cac7-bae8-4949-931a-7a1d4ddefc74)
Known problems: 
- some controls border should be repainted
- some windows/dialogs not repainted
- disabled menu items painted as disabled, even after enabling them
Probably would replace dark mode with custom mode, with color picker

**Comment by TarekLP** [comment]
this looks so good, although you should make the ability to choose own colours and export thems (if possible of course and not too time wasting)

**Comment by Duude92** [comment]
Fixed DropDownMenuItem rendering
